### DISCUSSION TITLE
Connect Thread Network to Blockchain Network(Remote Server)

### DISCUSSION THREAD

**Discussion by vipindas1984** [openthread/openthread#8458] [Q&A] [answered]
Hi, 

I am preparing a quick demo to connect my nRF5340 DK to a remote server(Blockchain Network) via internet. I setup OTBR using RPi4 and nRF52840 dongle by thinking that Router will forward the RPC  message back and forth from Thread End Device(nRF5340DK). 

- Does OpenThread supports sending RPC messages. 
- Is there any payload limitation in sending the messages. There might be a chance my application may send the message of size (Max 1K) and receive the message size(16K). 

Is there any existing example which shows or explain how to connect to remote server using UDP. 

Is it possible to use my nRF5340 DK (flashed with CLI sample) instead of nRF52840 dongle(flashed with RCP sample). This question may be foolish one but my requirement at this moment is to just send/receive messages(RPC)  from Remote Server. 

Regards

Vipin Das

**Comment by jwhui** [comment, marked as answer]
> Does OpenThread supports sending RPC messages.

OpenThread transports any IPv6 datagram.

> Is there any payload limitation in sending the messages. There might be a chance my application may send the message of size (Max 1K) and receive the message size(16K).

OpenThread supports the IPv6 minimum MTU of 1280 bytes. Note, however, that sending large datagrams requires fragmentation, which can impact overall reliability and latency.

> Is there any existing example which shows or explain how to connect to remote server using UDP.

[src/cli/README_UDP.md](https://github.com/openthread/openthread/blob/main/src/cli/README_UDP.md)

> Is it possible to use my nRF5340 DK (flashed with CLI sample) instead of nRF52840 dongle(flashed with RCP sample).

Yes.

**Reply by vipindas1984** [reply]
Hi @jwhui Jonathan,

Thanks for your quick reply. 

Further to my 4th question, I connected my nRF5340DK(flashed with nrf Thread:CLI sample) to RPi, just to replicate the same setup as previous ie OTBR with nRF52840 dongle. But by doing so and after checking the otbr-agent status I could see below result. 

`pi@pi:~ $ sudo service otbr-agent status
● otbr-agent.service - OpenThread Border Router Agent
     Loaded: loaded (/lib/systemd/system/otbr-agent.service; enabled; vendor preset: enabled)
     Active: activating (start-pre) since Wed 2022-11-30 14:15:24 CET; 2s ago
Cntrl PID: 4396 (systemctl)
      Tasks: 1 (limit: 4915)
        CPU: 57ms
     CGroup: /system.slice/otbr-agent.service
             └─4396 systemctl start mdns.service

Nov 30 14:15:24 pi systemd[1]: Starting OpenThread Border Router Agent...
Nov 30 14:15:26 pi systemd[1]: Started OpenThread Border Router Agent.
pi@pi:~ $ 
pi@pi:~ $ 
pi@pi:~ $ 
pi@pi:~ $ 
pi@pi:~ $ 
pi@pi:~ $ 
pi@pi:~ $ 
pi@pi:~ $ 
pi@pi:~ $ sudo service otbr-agent status
● otbr-agent.service - OpenThread Border Router Agent
     Loaded: loaded (/lib/systemd/system/otbr-agent.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Wed 2022-11-30 14:16:07 CET; 3s ago
    Process: 4430 ExecStartPre=service mdns start (code=exited, status=0/SUCCESS)
    Process: 4436 ExecStart=/usr/sbin/otbr-agent $OTBR_AGENT_OPTS (code=exited, status=6)
   Main PID: 4436 (code=exited, status=6)
        CPU: 75ms
pi@pi:~ $ 

pi@pi:~ $ sudo service otbr-agent status
● otbr-agent.service - OpenThread Border Router Agent
     Loaded: loaded (/lib/systemd/system/otbr-agent.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2022-11-30 14:24:24 CET; 1s ago
    Process: 4874 ExecStartPre=service mdns start (code=exited, status=0/SUCCESS)
   Main PID: 4880 (otbr-agent)
      Tasks: 1 (limit: 4915)
        CPU: 72ms
     CGroup: /system.slice/otbr-agent.service
             └─4880 /usr/sbin/otbr-agent -I wpan0 -B eth0 spinel+hdlc+uart:///dev/ttyACM0?uart-baudrate=1000000 trel://eth0

Nov 30 14:24:21 pi systemd[1]: Starting OpenThread Border Router Agent...
Nov 30 14:24:24 pi systemd[1]: Started OpenThread Border Router Agent.
Nov 30 14:24:24 pi otbr-agent[4880]: [NOTE]-AGENT---: Backbone interface: eth0
Nov 30 14:24:24 pi otbr-agent[4880]: [NOTE]-AGENT---: Running 0.3.0-32a7779
Nov 30 14:24:24 pi otbr-agent[4880]: [NOTE]-AGENT---: Thread version: 1.3.0
Nov 30 14:24:24 pi otbr-agent[4880]: [NOTE]-AGENT---: Thread interface: wpan0
Nov 30 14:24:24 pi otbr-agent[4880]: [NOTE]-AGENT---: Radio URL: spinel+hdlc+uart:///dev/ttyACM0?uart-baudrate=1000000
Nov 30 14:24:24 pi otbr-agent[4880]: [NOTE]-AGENT---: Radio URL: trel://eth0
Nov 30 14:24:24 pi otbr-agent[4880]: [NOTE]-ILS-----: Infra link selected: eth0
Nov 30 14:24:24 pi otbr-agent[4880]: [INFO]-NCP-----: OpenThread log level changed to 4`

I think the service is getting restarted continuously. Is this problem happening because of missing RCP chip?  

Continuation to my previous question, Would I be able to setup a OTBR with nRF5340DK(nrf Thread CLI sample) and Connect to internet directly from the Router itself. 

Regards

Vipin Das

**Reply by jwhui** [reply]
OTBR requires running the RCP image on the dongle.

**Reply by vipindas1984** [reply]
Thanks Jonathan. 

It's clear now. 

Thanks 

Vipin Das
